base/intfuncs.jl

```diff
@@ -174,8 +173,8 @@ function power_by_squaring(x_, p::Integer)
     elseif p == 2
         return x*x
     elseif p < 0
-        x == 1 && return copy(x)
-        x == -1 && return iseven(p) ? one(x) : copy(x)
+        isone(x) && return copy(x)
+        isone(-x) && return iseven(p) ? one(x) : copy(x)
         throw_domerr_powbysq(p)
```

Comment:
The error message printed in `throw_domerr_powbysq` could perhaps be updated to apply to integer matrices as well? Easiest would probably be to add a second method that explains the situation for integer matrices. Then this line could instead be
```julia
throw_domerr_powbysq(x, p)
```
and dispatch on `x`.